src/WebJobs.Extensions.DurableTask/Listener/TaskActorShim.cs

```diff
@@ -0,0 +1,323 @@
+using DurableTask.Core.Middleware;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Implements the actor scheduler as a looping orchestration.
+    /// There is one such orchestration per actor.
+    /// The orchestration terminates if the actor is deleted and idle.
+    /// The orchestration calls ContinueAsNew when it is idle, but not deleted.
+    /// </summary>
+    internal class TaskActorShim : TaskCommonShim
+    {
+        private readonly DurableActorContext context;
+
+        private TaskCompletionSource<object> shutDownSignal
+            = new TaskCompletionSource<object>();
+
+        private Task lastStartedOperation = Task.CompletedTask;
+
+        private Dictionary<Guid, Task> currentReentrantRequests = new Dictionary<Guid, Task>();
+
+        private int remainingReceiveLimit = 20; // TODO make configurable and/or compute based on state size
+
+        public TaskActorShim(DurableTaskExtension config, string schedulerId)
+            : base(config)
+        {
+            this.SchedulerId = schedulerId;
+            this.ActorId = GetActorIdFromSchedulerId(schedulerId);
+            this.context = new DurableActorContext(config, this.ActorId);
+        }
+
+        public override DurableCommonContext Context => context;
+
+        public string SchedulerId { get; private set; }
+
+        public ActorId ActorId { get; private set; }
+
+        public static string GetSchedulerIdFromActorId(ActorId actor)
+        {
+            return $"@{@actor.ActorClass}@{actor.ActorKey}";
+        }
+
+        public static ActorId GetActorIdFromSchedulerId(string schedulerId)
+        {
+            var pos = schedulerId.IndexOf('@', 1);
+            var actorClass = schedulerId.Substring(1, pos - 1);
+            var actorKey = schedulerId.Substring(pos + 1);
+            return new ActorId(actorClass, actorKey);
+        }
+
+        public override RegisteredFunctionInfo GetFunctionInfo()
+        {
+            FunctionName actorFunction = new FunctionName(this.Context.FunctionName);
+            return this.Config.GetActorInfo(actorFunction);
+        }
+
+        public override string GetStatus()
+        {
+            if (this.lastStartedOperation.IsCompleted)
+            {
+                return "Available";
+            }
+            else
+            {
+                if (this.context.State.Queue != null)
+                {
+                    return $"Busy ({this.context.State.Queue.Count} Queued)";
+                }
+                else
+                {
+                    return "Busy";
+                }
+            }
+        }
+
+        private void SignalShutdown()
+        {
+            MethodInfo trySetResult = typeof(TaskCompletionSource<object>).GetMethod("TrySetResult");
+            trySetResult.Invoke(this.shutDownSignal, new[] { (object)null });
```

Comment:
Why do we need reflection here? It seems to me we can just call `TrySetResult` directly, like we are in the line below.

Reply:
This is a strange trick I copied from your code: calling using reflection forces the completion to be executed inline, i.e. the same thread will continue at the await. Not sure why it works, but I was assuming you knew. I actually meant to ask you about this.

(I tried without that but ran into trouble because the DTFx framework expects all the orchestrator actions to be produced before RaiseEvents returns).

Of course there is still a bug here, I should not be doing this twice, the last line is superfluous.